pkg/kubelet/cm/dra/manager.go

```diff
@@ -149,7 +150,7 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 			}
 			batches[pluginName] = append(batches[pluginName], claim)
 		}
-		claimInfos[resourceClaim.UID] = claimInfo
+		claimInfos[resourceClaim.UID] = &claimInfoCopy
```

Comment:
Remind me what the difference between this claimInfos slice and the claimInfo cache is. Why is this not just a call to `m.cache.add(&claimInfoCopy)` (or `update()` as I suggested above)?

Reply:
Objects in this map(new claim infos or copies of the cached ones) are not cached. They're used to to perform batched calls of the Node[Un]PrepareResources GRPC. They're added to the cache only after calls succeed. BTW, this can be a valid fix for https://github.com/kubernetes/kubernetes/issues/116411.

Reply:
Ack, this seems correct then. I'd need to look closer at #116411 to see how this is related